### PR TITLE
Removes defunct resource editing functionality

### LINKED ISSUE #523: Remove "edit" link from Content portion of detail display
This shows up for any object that is editable -- registered or opened -- with content.  See https://argo-dev.stanford.edu/catalog/druid:qs553rb8234

screen shot 2016-02-16 at 10 35 07 am

### PR DESCRIPTION
**Pull request by mejackreed**
This commit removes legacy and non-working resource editing functionality. This functionality is broken and if needed, should be rewritten accordingly with updated patterns.

Closes #523 

I hope this can also reduce some complexity and comments on #516 and the various APO related changes PR's by @darrenleeweber .

### CODE REVIEW

**Comment by darrenleeweber** [comment]
Well I'm glad we attended to that XXX spec and resolved to remove all this.  I suggest a patch to keep all this stuff get thrown into a branch that may go stale.

**Comment by mejackreed** [comment]
Can you elaborate on your suggestion about the patch and branch? I'm not sure I follow a :100:%

**Comment by darrenleeweber** [comment]
When this thing is good to go, merge it, create a new branch, reset and commit to that branch.  I guess that will thereby keep everything thrown away in a tidy form at the time it was thrown away.  With a branch name on it, it should be easier to find than trolling through commits?  Anyhow, no big deal.

**Comment by darrenleeweber** [comment]
:+1: (although I don't know much about these features and whether they would be easy to complete rather than throw them away).

**Comment by mejackreed** [comment]
Ah gotcha good call. Just created #566 which references the SHA and this PR so we can dig it up later on.

File: app/views/catalog/_show_sections/_default_contents.html.erb [hunk 1]
```diff
@@ -25,10 +25,6 @@
           content_ds.ng_xml.xpath('/contentMetadata/resource').each_with_index do |resource, index| %>
             <li class="resource">
               <span class="label">Resource (<%=index+1%>)</span> <%= resource['type']%>
-              <%if can_manage && version_open
-                edit_resource_link="/items/#{params[:id]}/resource/?edit=true&resource=#{resource['id']}"%>
-                <%= link_to "(edit)", edit_resource_link, :data => {ajax_modal: 'trigger'} %>
-              <%end%>
```
Comment (drh-stanford) [inline app/views/catalog/_show_sections/_default_contents.html.erb]: The `version_open = can_close_version? object.pid` statement can also be deleted, right?

Comment (mejackreed) [inline app/views/catalog/_show_sections/_default_contents.html.erb]: Thanks for catching that @drh-stanford ! Just pushed a change removing that.

**Comment by drh-stanford** [comment]
This PR strictly removes code from the unused resource (file) management UIs.

**Pull request merged**